library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsMediaChunk.java

```diff
@@ -400,6 +400,15 @@ public void publish() {
     isPublished = true;
   }
 
+  /**
+   * If their are any samples loaded in this chunk.
+   *
+   * @return true if samples were loaded
+   */
+  boolean hasSamples() {
+    return ! hasGapTag;   // TOOD - could expand upon this to be more accurate, but for sure GAP has none
+  }
+
```

Comment:
In theory, we could have the parsing state with committed samples as well maybe, thread safety, as this is called from the playback thread.